src/node_report.cc

```diff
@@ -369,12 +366,28 @@ void TriggerNodeReport(Isolate* isolate, DumpEvent event, const char* message, c
  * Function to print Node.js version, OS version and machine information
  *
  ******************************************************************************/
-static void PrintVersionInformation(FILE* fp) {
+static void PrintVersionInformation(FILE* fp, Isolate* isolate) {
 
   // Print Node.js and deps component versions
-  fprintf(fp, "\nNode.js version: %s\n", NODE_VERSION);
-  fprintf(fp, "(v8: %s, libuv: %s, zlib: %s, ares: %s)\n",
-          V8::GetVersion(), uv_version_string(), ZLIB_VERSION, ARES_VERSION_STR);
+  fprintf(fp, "\nprocess.versions = {\n");
+  {
+    Nan::MaybeLocal<Value> process = Nan::Get(
+      isolate->GetCurrentContext()->Global(),
+      Nan::New<String>("process").ToLocalChecked());
+    Local<Value> versions = Nan::Get(process.ToLocalChecked().As<Object>(),
+      Nan::New<String>("versions").ToLocalChecked()).ToLocalChecked();
+    Local<Object> versionsObj = versions.As<Object>();
+    Local<v8::Array> keys = Nan::GetOwnPropertyNames(versionsObj)
+      .ToLocalChecked();
+    for (uint32_t i=0, nKeys = (*keys)->Length(); i<nKeys; i++) {
+      Local<Value> keyVal = Nan::Get(keys.As<Object>(), i).ToLocalChecked();
+      Local<Value> valVal = Nan::Get(versionsObj, keyVal).ToLocalChecked();
+      Nan::Utf8String name(keyVal);
+      Nan::Utf8String version(valVal);
+      fprintf(fp, "  %s: %s\n", *name, *version);
+    }
+  }
+  fprintf(fp, "}\n");
```

Comment:
I would capture it at the earliest opportunity, that will let you deal with errors more gracefully.  For example, if user JS code (however unwise) did the below, your code would assert and abort.
```js
Object.defineProperty(process, 'versions', {get() { throw 'boom' }});
```
This particular example is a little farfetched but buggy code clobbering `process` is not unheard of.